public/js/test/mochitest/browser_dbg-editor-pending-select.js

```diff
@@ -0,0 +1,34 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+// Tests that the editor will select the right source and line even if
+// the source doesn't exist yet. This is the workflow that happens
+// when you click on a location in the console and the debugger isn't
+// initialized yet.
+
+function openConsole() {
+  return openNewTabAndToolbox(
+    EXAMPLE_URL + url,
+    "webconsole"
+  );
+}
+
+add_task(function* () {
+  const dbg = yield initDebugger("doc-scripts.html");
+  const { selectors: { getSelectedSource }, getState } = dbg;
+  const sourceUrl = EXAMPLE_URL + "code-long.js";
+
+  // The source probably hasn't been loaded yet, and using
+  // `selectSourceURL` will set a pending request to load this source
+  // and highlight a specific line.
+  dbg.actions.selectSourceURL(sourceUrl, { line: 66 });
```

Comment:
its mostly a nit because it *doesn't* make a big difference in this case. We've done a good job of keeping the test fairly high level.